### PR TITLE
LPS-52817 DM thumbnail colors are incorrect for special image

### PR DESCRIPTION
**Pull request by zsolah**
Hi,

Please review.
I have added a new image IO library to accomplish some problematic image read, and simplify the code. I have tested the test images from previous related LPS-es they worked as expected. This change can be also regarded as an improvement. This new library (TwelveMonkeys) is better and active on github compared to the previous one.

Regards,
Zsolt

### CODE REVIEW

**Comment by IstvanD** [comment]
Forwarded to https://github.com/sergiogonzalez/liferay-portal/pull/2658

File: portal-impl/test/integration/com/liferay/portal/image/ImageToolImplTest.java [hunk 1]
```diff
@@ -163,9 +160,10 @@ protected void read(String fileName) throws Exception {
 
 		Assert.assertNotNull(expectedImage);
 
-		Raster raster = expectedImage.getData();
+		DataBufferByte expectedDataBufferByte =
```
Comment (zsolah) [inline portal-impl/test/integration/com/liferay/portal/image/ImageToolImplTest.java]: The modification of this test can be kept.

File: portal-impl/src/com/liferay/portal/image/ImageToolImpl.java [hunk 2]
```diff
@@ -472,32 +388,28 @@ public ImageBag read(byte[] bytes) throws IOException {
...
 				imageInputStream);
 
-			while ((renderedImage == null) && iterator.hasNext()) {
+			while (iterator.hasNext()) {
 				ImageReader imageReader = iterator.next();
 
 				imageReaders.offer(imageReader);
 
-				try {
+				if (firstImageReader) {
```
Comment (zsolah) [inline portal-impl/src/com/liferay/portal/image/ImageToolImpl.java]: firstImageReader is obsolete, if the revert takes place

File: portal-impl/src/com/liferay/portal/image/ImageToolImpl.java [hunk 3]
```diff
@@ -472,32 +388,28 @@ public ImageBag read(byte[] bytes) throws IOException {
...
+			boolean firstImageReader = true;
+
 			imageInputStream = ImageIO.createImageInputStream(
 				new ByteArrayInputStream(bytes));
 
 			Iterator<ImageReader> iterator = ImageIO.getImageReaders(
 				imageInputStream);
 
-			while ((renderedImage == null) && iterator.hasNext()) {
+			while (iterator.hasNext()) {
```
Comment (zsolah) [inline portal-impl/src/com/liferay/portal/image/ImageToolImpl.java]: the (renderedImage == null) condition makes sense here

File: lib/versions.xml [hunk 4]
```diff
@@ -2596,7 +2680,7 @@
 				<file-name>lib/portal/netty-all.jar</file-name>
 				<version>4.0.23</version>
 				<project-name>Netty</project-name>
-				<project-url>http://netty.io</project-url>
+				<project-url>http://netty.io/</project-url>
```
Comment (zsolah) [inline lib/versions.xml]: This revert was also out of scope

File: lib/versions.html [hunk 5]
```diff
@@ -1393,7 +1450,7 @@
 </tr>
 			
 <tr>
-<td nowrap>lib/portal/netty-all.jar</td><td nowrap>4.0.23</td><td nowrap><a href="http://netty.io">Netty</a></td><td nowrap><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a>
+<td nowrap>lib/portal/netty-all.jar</td><td nowrap>4.0.23</td><td nowrap><a href="http://netty.io/">Netty</a></td><td nowrap><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a>
```
Comment (zsolah) [inline lib/versions.html]: Out of scope these 2 libraries

**Pull request closed without merging**